### PR TITLE
Feature: Lazily loading themes to improve startup time

### PR DESCRIPTION
**Pull request by hez2010**
- Enable dynamic PGO
- Postpone loading other themes to the time user open the theme settings
- Use x:Bind instead of Binding where possible

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

**Screenshots (optional)**
Add screenshots here.

### CODE REVIEW

**Comment by yaichenbaum** [comment]
> Enable dynamic PGO

Are there any benefits from this change?

File: src/Files.App/Views/SettingsPages/Appearance.xaml [hunk 1]
```diff
@@ -63,18 +64,21 @@
...
-					VerticalAlignment="Bottom"
-					AutomationProperties.Name="{helpers:ResourceString Name=SettingsThemesLearnMoreButton/AutomationProperties/Name}"
-					Background="Transparent"
-					BorderBrush="Transparent"
-					Click="ThemesLearnMoreButton_Click"
-					ToolTipService.ToolTip="{helpers:ResourceString Name=SettingsThemesLearnMoreButton/ToolTipService/ToolTip}">
-					<FontIcon FontSize="14" Glyph="&#xE946;" />
-				</Button>
+				<StackPanel Orientation="Horizontal">
+					<ProgressRing x:Name="AppThemeLoading" Height="32" Width="32" Padding="8" VerticalAlignment="Bottom" Visibility="Collapsed" IsIndeterminate="True" />
```
Comment (yaichenbaum) [inline src/Files.App/Views/SettingsPages/Appearance.xaml:68]: Should this have an automation name for screen readers?

**Comment by yaichenbaum** [comment]
This seems to break custom themes, I think we need to load the current selected theme at startup.

**Comment by hez2010** [comment]
> > Enable dynamic PGO
> 
> Are there any benefits from this change?

It can instrument methods and optimize them using profile data at the runtime.

**Comment by yaichenbaum** [comment]
@hez2010 can you see if changing the custom theme adds the selected theme to the end of the list for you?
image

File: src/Files.App/Views/SettingsPages/Appearance.xaml (hunk 1 above)
Comment (hez2010) [reply src/Files.App/Views/SettingsPages/Appearance.xaml:68]: Is there an existing localized string for such loading indicator?

Comment (yaichenbaum) [reply src/Files.App/Views/SettingsPages/Appearance.xaml:68]: Not yet, let's add one for "Loading themes".

**Comment by hez2010** [comment]
> @hez2010 can you see if changing the custom theme adds a duplicate theme? image

Weird. I think I have guarded them here: 
https://github.com/files-community/Files/pull/10329/files#diff-d43965edb43d80794b663802fa2b2a4feb4f039c29d8107c2ed2ef42ca5caa4dR97

File: src/Files.App/Helpers/ExternalResourcesHelper.cs [hunk 2]
```diff
@@ -71,13 +72,13 @@ public async Task LoadOtherThemesAsync()
...
-			foreach (var file in (await folder.GetFilesAsync()).Where(x => x.FileType == ".xaml"))
+			foreach (var file in (await folder.GetFilesAsync()).Where(x => string.Equals(x.FileType, ".xaml", StringComparison.InvariantCultureIgnoreCase)))
 			{
-				if (!Themes.Exists(t => t.AbsolutePath == file.Path))
+				if (!Themes.Any(t => t.AbsolutePath == file.Path))
 				{
 					Themes.Add(new AppTheme()
 					{
-						Name = file.Name.Replace(".xaml", "", StringComparison.Ordinal),
+						Name = file.Name[..^5],
```
Comment (yaichenbaum) [inline src/Files.App/Helpers/ExternalResourcesHelper.cs:81]: What does `^5` do? I don't think I've seen this before.

File: src/Files.App/Views/SettingsPages/Appearance.xaml.cs [hunk 3]
```diff
@@ -56,5 +56,15 @@ private async void ViewModel_PropertyChanged(object sender, System.ComponentMode
...
             }
         }
+
+        private void AppThemeSelectionGridView_Loaded(object sender, RoutedEventArgs e)
+        {
+            AppThemeLoading.Visibility = Visibility.Visible;
+            DispatcherQueue.TryEnqueue(async () =>
+            {
+                await App.ExternalResourcesHelper.LoadOtherThemesAsync();
+                AppThemeLoading.Visibility = Visibility.Collapsed;
```
Comment (yaichenbaum) [inline src/Files.App/Views/SettingsPages/Appearance.xaml.cs:66]: Would it be better to create a property in the view model for this?

File: src/Files.App/Helpers/ExternalResourcesHelper.cs (hunk 2 above)
Comment (hez2010) [reply src/Files.App/Helpers/ExternalResourcesHelper.cs:81]: It's trimming the last 5 entries from a list (here is trimming the last 5 char from a string).

File: src/Files.App/Views/SettingsPages/Appearance.xaml.cs (hunk 3 above)
Comment (hez2010) [reply src/Files.App/Views/SettingsPages/Appearance.xaml.cs:66]: You're right.

**Comment by hez2010** [comment]
> Need debug to investigate.

Resolved.

File: src/Files.App/Views/SettingsPages/Appearance.xaml (hunk 1 above)
Comment (hez2010) [reply src/Files.App/Views/SettingsPages/Appearance.xaml]: What's the workflow of adding a new string now?

Comment (hez2010) [reply src/Files.App/Views/SettingsPages/Appearance.xaml]: I've added one `SettingsThemesLoadingProgressRing/AutomationProperties/Name`.

File: src/Files.App/Views/SettingsPages/Appearance.xaml [hunk 4]
```diff
@@ -52,29 +52,33 @@
...
-					VerticalAlignment="Bottom"
-					AutomationProperties.Name="{helpers:ResourceString Name=SettingsThemesLearnMoreButton/AutomationProperties/Name}"
-					Background="Transparent"
-					BorderBrush="Transparent"
-					Click="ThemesLearnMoreButton_Click"
-					ToolTipService.ToolTip="{helpers:ResourceString Name=SettingsThemesLearnMoreButton/ToolTipService/ToolTip}">
-					<FontIcon FontSize="14" Glyph="&#xE946;" />
-				</Button>
+				<StackPanel Orientation="Horizontal">
+					<ProgressRing Height="32" Width="32" Padding="8" VerticalAlignment="Bottom" AutomationProperties.Name="{helpers:ResourceString Name=SettingsThemesLoadingProgressRing/AutomationProperties/Name}" Visibility="{x:Bind ViewModel.LoadingTheme, Mode=OneWay}" IsIndeterminate="True" />
```
Comment (yaichenbaum) [inline src/Files.App/Views/SettingsPages/Appearance.xaml:68]:
```suggestion
					<ProgressRing Height="32" Width="32" Padding="8" VerticalAlignment="Bottom" AutomationProperties.Name="{helpers:ResourceString Name=LoadingThemes}" Visibility="{x:Bind ViewModel.LoadingTheme, Mode=OneWay}" IsIndeterminate="True" />
```

File: src/Files.App/Views/SettingsPages/Appearance.xaml (hunk 1 above)
Comment (yaichenbaum) [reply src/Files.App/Views/SettingsPages/Appearance.xaml]: If you add a string to the en-us resource file called "LoadingThemes", Crowdin will handle the other resource files after this PR is merged.

File: src/Files.App/Views/SettingsPages/Appearance.xaml (hunk 4 above)
Comment (yaichenbaum) [reply src/Files.App/Views/SettingsPages/Appearance.xaml:68]: We could probably also use x:Load and bind to a boolean property.

**Review by yaichenbaum — APPROVED** [review]
LGTM

**Pull request merged**